### DISCUSSION TITLE
[Bug] Thrift 0.14.1, traceId will be duplicated

### DISCUSSION THREAD

**Discussion by xiaohei88** [apache/skywalking#11295] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

If I use Thrift 0.14.1, there will be a duplicate traceId issue.

### What you expected to happen

I hope traceContext will not leak and traceId will not be duplicated.

### How to reproduce

use Thrift 0.14.1.

### Anything else

In version 0.14.1, the following code conversion for EnhancedInstance will fail. So SkyWalkingDynamicField is empty
```java
TServiceClientInterceptor.java
 Object dynamicField = ((EnhancedInstance) protocol.getTransport()).getSkyWalkingDynamicField();
```
ContextManager. stopSpan(); Will fail, leading to leakage.
```java
TServiceClientReceiveInterceptor.java

 if (Objects.nonNull(objInst.getSkyWalkingDynamicField())) {
            ContextManager.stopSpan();
        }
```
I'm not sure if it's a bug. If anyone encounters this issue, you can pay attention.

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Please debug the plugin, and find out why this is empty. 
Global individuals contribute our plugins, we don't know much about the details of every plugin.

**Comment by wu-sheng** [comment]
The issue is only for tracking issues having owner, like https://github.com/apache/skywalking/issues/11293. It is proposed by someone is going to fix on their own.

**Comment by pengxingjia** [comment]
skywalking 9.0.0 thrift 0.14.1

What happened?
A large number of NoopExitSpans are being generated in the application stack and not being released, resulting in a memory overflow within the application.

Object dynamicField = ((EnhancedInstance) protocol.getTransport()).getSkyWalkingDynamicField();
The exception message is
java.lang.ClassCastException: org.apache.thrift.transport.TFramedTransport cannot be cast to org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance

try {
Object dynamicField = ((EnhancedInstance) protocol.getTransport()).getSkyWalkingDynamicField();
objInst.setSkyWalkingDynamicField(Objects.isNull(dynamicField) ? "UNKNOWN" : dynamicField);
}catch (Exception ex){
objInst.setSkyWalkingDynamicField("UNKNOWN");
}


Catching this here would eliminate the error, but I'm not sure if it would cause other issues

**Reply by wu-sheng** [reply]
What exception do you catch actually?

**Reply by pengxingjia** [reply]
`Object dynamicField = ((EnhancedInstance) protocol.getTransport()).getSkyWalkingDynamicField();`

the following code conversion for EnhancedInstance will fail. A large number of NoopExitSpans are being generated in the application stack and not being released
java.lang.ClassCastException: org.apache.thrift.transport.TFramedTransport cannot be cast to org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance

**Reply by wu-sheng** [reply]
> java.lang.ClassCastException: org.apache.thrift.transport.TFramedTransport cannot be cast to org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance

This means `TFramedTransport` doesn't be instrumented. `try/catch` doesn't resolve the root cause. 
The good practice should be, find why TFramedTransport is missed to instrument and add that to keep this class convert successfully.
